### DISCUSSION TITLE
Font set documentation

### DISCUSSION THREAD

**Discussion by masinter** [Interlisp/medley#180] [Q&A]
Since the discussion has mainly been about font set documentation and not about MAKEINIT, I started a new Discussion #185

**Comment by johnwcowan** [comment]
Is there documentation somewhere on what ^A through ^F actually mean?

**Reply by masinter** [reply]
i'm sure it's in the sun user's guide or medley primer or lisp manual but i haven't looked yet

**Reply by Anzus** [reply]
https://github.com/Interlisp/medley/blob/master/docs/primer/02-TYPING-SHORTCUTS.pdf has a few of them, but not in a convenient table form

**Reply by ecraven** [reply]
I'd be interested in that too, they are all over the source files, but don't seem to directly influence display in SEdit.

**Reply by nbriggs** [reply]
I'm guessing you're talking about the ^F^[A-F] in the sources?  They're (indirectly) mentioned in FONTPROFILE

```
                       (* ;; "Fonts for prettyprinting")
                          (COMMENTFONT LITTLEFONT) (* ; "for comments ")
                          (PRETTYCOMFONT BOLDFONT)  (* ; " for words being defined")
                          (CLISPFONT BOLDFONT) (* ; " for keywords & CLISP")
                          (SYSTEMFONT DEFAULTFONT) (* ; " for %"system%" words(?)")
                          (LAMBDAFONT BIGFONT) (* ; "for words being defined")
                          (USERFONT BOLDFONT) (* ; " for %"user%" defined words")
```
One should be able to map these COMMENT/PRETTYCOM/CLISP/SYSTEM/LAMBDA/USER back to the ^F^[A-F] markup.

**Comment by nbriggs** [comment]
It seems to come from `PRINTCOMSTRAN` on file MACROS, at least as used for prettyprinting source files --
- ^F^A (font 1) default font (on a display this would be Gacha 10 pt, in print Gacha/Terminal 8 pt)
- ^F^B (font 2) prettycom font and user font, a bold font, Helvetica 10 bold on a display, in print Helvetica 8 bold)
- ^F^C (font 3) littlefont, Helvetica 8 on display, in print Helvetica 6 italic
- ^F^D (font 4) lambda font, bigfont, Helvetica 12 bold on display, in print Modern 10 bold
- ^F^E (font 5) menu font, Helvetica 10 on display, no corresponding font in print
- ^F^F (font 6) comment (and tiny) font, Gacha 8 on display, in print Gacha/Terminal 6
- ^F^G (font 7) textfont, unlikely in prettyprinted files, TimesRoman 10 on display, in print Classic 10 (a serif font)

There seems to be a subscript and a superscript font as well -- I don't think these can show up in prettyprinted files.  ^F (character code 6) is the font change introducer, the next byte is an index into a FONTSET which gives you a FONTCLASS, which gives you the display, press, interpress, and "other" fonts.  The "other" fonts is a ALIST which may include, for example, ((POSTSCRIPT TERMINAL 6)), for the "tinyfont".  The fonts I quote above are from the (FONTSET 'STANDARD) -- you can change the fonts through FONTSET names including BIGGER, MEDIUM, STANDARD, BIG, HUGE and you can create your own if you desire which would change the presentation through things like the "PP" command.

**Reply by masinter** [reply]
Where should this be documented? Please indicate which:
1. Just here
2. in a Discussion FAQ
2. In the wiki (a) in the intro (b) somewhere else
3. In a revision of the Primer
4. In a revision of the IRM
5. somewhere else
you can choose more than one.

**Reply by nbriggs** [reply]
It's actually already documented in the IRM I/O volume, pages 24-20 through 24-23, if you know where to look.

**Reply by ecraven** [reply]
An entry in the wiki for beginners on how to change the default font to bigger [I think (FONTSET 'BIGGER) or even (FONTSET 'HUGE) ] might be helpful, on the last live-stream someone asked whether it was possible to increase the font size.

Is there a way to re-load the prompt window and the small top bar after these fontset changes, so they also appear bigger?

**Comment by ecraven** [comment]
> It's actually already documented in the IRM I/O volume, pages 24-20
> through 24-23, if you know where to look.

Which version is that? The DInfo that comes with Medley only goes up to 24-18 :-/

Thanks for any help with this!
Greetings,
Peter

**Comment by nbriggs** [comment]
On the website -- https://interlisp.org <https://interlisp.org/> at Interlisp Reference Manual <https://interlisp.org/docs/IRM.pdf>

-- Nick

**Reply by ecraven** [reply]
Is that file generated from some sort of sources? I've noticed some typos that I could contribute a fix for.

**Comment by nbriggs** [comment]
BTW -- Page numbers may be different in the DInfo version -- look for the documentation for "PRINTOUT"

**Reply by masinter** [reply]
Following RFC 8118;   https://interlisp.org/docs/IRM.pdf#page=580

**Reply by ecraven** [reply]
For reference, it is 25-6 in the DInfo version.
![maim](https://user-images.githubusercontent.com/215510/107849630-cdfc9200-6dfc-11eb-9847-abcf356b6b1b.png)

**Comment by ecraven** [comment]
> BTW -- Page numbers may be different in the DInfo version -- look for the documentation for "PRINTOUT"

Thanks a lot! With TYPE on the exec, I found out the following
(apologies if all this is already known):

TYPE in an Interlisp Exec will print things nicely marked up. You can
use it to print files. I created a file containing different sorts of
markup, see the results attached as an image.

- ^A is the normal font
- ^B is the same size, but bold (and a bit more condensed)
- ^C is a smaller font
- ^D is a larger bold font
- ^E is a wider sans-serif font
- ^F is an even smaller font (or more condensed?)
- ^G is a narrower serif font

Thanks for all the hints on how to find this out, I'll go looking for
the exact places where these are *printed* later ;) I tried up to ^K,
but nothing more interesting happened ;)

![maim](https://user-images.githubusercontent.com/215510/107819749-f1cac400-6d79-11eb-9565-3c735fb48e49.png)

**Comment by nbriggs** [comment]
> On Feb 12, 2021, at 12:32 PM, ecraven <notifications@github.com> wrote:
> 
> 
> > BTW -- Page numbers may be different in the DInfo version -- look for the documentation for "PRINTOUT"
> 
> Thanks a lot! With TYPE on the exec, I found out the following
> (apologies if all this is already known):
> 
> TYPE in an Interlisp Exec will print things nicely marked up. You can
> use it to print files. I created a file containing different sorts of
> markup, see the results attached as an image.
> 
> - ^A is the normal font
> - ^B is the same size, but bold (and a bit more condensed)
> - ^C is a smaller font
> - ^D is a larger bold font
> - ^E is a wider sans-serif font
> - ^F is an even smaller font (or more condensed?)
> - ^G is a narrower serif font

They're not prescriptive in the way you're suggesting... IF you happen to be using the "STANDARD" font set and haven't changed any components of it (which you can trivially do) THEN those would be the results.  However "^F^C" doesn't mean "a smaller font", it means "Font #3, of the current font set, whatever that might be at the time you display the text" (it's dynamically bound).   You can look at the pretty printer to see how it uses font numbers 1..6, (which were the names that I mentioned, such as lambda, comment)

**Reply by nbriggs** [reply]
Try (IL exec): `(FONTSET 'BIGGER)` and then type your test output again.  You can make your own fontset as well, with whatever fonts you want in slots 1..n (which will allow you to use more than the 7 in the STANDARD fontset).

**Comment by masinter** [comment]
I think this discussion belongs in the Wiki with a pointer from the Medley repo README (where it talks about what files are where)
